### PR TITLE
[WIP] @gndclouds/community page device padding

### PR DESCRIPTION
**Pull request by gndclouds**
Todo: 
- The GitHub Issues importing is 3 components, I need to refactor it into the page.
- Animated Terminal Stops halfway through annotation.
- Some redundant CSS to clean

### CODE REVIEW

File: components/core/CodeTerminal.js [hunk 1]
```diff
@@ -0,0 +1,116 @@
+import React, { useState, useEffect } from "react";
+import * as Constants from "~/common/constants";
+import * as System from "~/components/system";
+
+import { css, keyframes } from "@emotion/react";
+
+const blinkCursor = keyframes`
+0% {opacity: 0;}
```
Comment (jimmylee) [inline components/core/CodeTerminal.js:8]: you'll want to fix formatting here.

File: components/core/CodeTerminal.js [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+50% {opacity: 1;}
+100% {opacity: 0;}
+`;
+
+const typewriter = keyframes`
+  0%,100% {width: 0;}
```
Comment (jimmylee) [inline components/core/CodeTerminal.js:14]: and here.

**Comment by jimmylee** [comment]
* `core/marketing/**/*` is where you should put the marketing components. 
* re-read your formatting, re-read your code
* rename and rewrite the slate group components? thats just copy & paste?

**Comment by jimmylee** [comment]
you have merge conflicts in this PR.

**Review by jimmylee — CHANGES_REQUESTED** [review]
Addressed in the comments

**Comment by gndclouds** [comment]
@jimmylee Should be good

File: components/core/marketing/CodeTerminal.js [hunk 3]
```diff
@@ -0,0 +1,113 @@
+import React, { useState, useEffect } from "react";
+import * as Constants from "~/common/constants";
+import * as System from "~/components/system";
+
+import { css, keyframes } from "@emotion/react";
+
+const blinkCursor = keyframes`
+0% {
```
Comment (jimmylee) [inline components/core/marketing/CodeTerminal.js:8]: formatting.

File: components/core/marketing/CodeTerminal.js [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+}
+50% {
+  opacity: 1;
+}
+100% {
+  opacity: 0;
+}
+`;
+const typewriter = keyframes`
+  0%,100% {
```
Comment (jimmylee) [inline components/core/marketing/CodeTerminal.js:19]: formatting

File: components/core/marketing/CodeTerminal.js [hunk 5]
```diff
@@ -0,0 +1,113 @@
...
+    width: 0;
+  }
+  20%, 80% {
+    width: 10.2em;
+  }
+`;
+const STYLES_ROOT = css`
```
Comment (jimmylee) [inline components/core/marketing/CodeTerminal.js:26]: new line between these things just like every other file in the codebase

File: components/core/marketing/CodeTerminal.js [hunk 6]
```diff
@@ -0,0 +1,113 @@
...
+  }
+  :nth-of-type(2) {
+    background: rgb(255, 189, 46);
+  }
+  :nth-of-type(3) {
+    background: rgb(39, 201, 63);
+  }
+`;
+const STYLES_WINDOW_BODY = css`
+  padding: 25px;
```
Comment (jimmylee) [inline components/core/marketing/CodeTerminal.js:90]: You should stick to multiples of 8, every thing in our product, marketing, is off of a pt4 grid (easier to think of it as base 8), which means multiples of 4, 8, and 12 are acceptable.

File: components/core/marketing/Issue.js [hunk 7]
```diff
@@ -0,0 +1,119 @@
...
+
+import { css } from "@emotion/react";
+
+const STYLES_SLATE_CARD_GROUP = css`
+  display: flex;
+  flex-wrap: wrap;
+  width: 100%;
+  margin-top: 48px;
+`;
+const STYLES_SLATE_CARD = css`
```
Comment (jimmylee) [inline components/core/marketing/Issue.js:12]: Formatting, and the name of these `STYLES_` don't make sense. Right?

File: components/core/marketing/Issue.js [hunk 8]
```diff
@@ -0,0 +1,119 @@
...
+const STYLES_SLATE_CARD_CTA_PARAGRAPH = css`
+  font-size: ${Constants.typescale.lvl2};
+  text-align: left;
+  @media (max-width: ${Constants.sizes.mobile}px) {
+    font-size: 1rem;
+  }
+`;
+export default class Issue extends Component {
+  render() {
+    const { title, id, labels, userName, url } = this.props;
```
Comment (jimmylee) [inline components/core/marketing/Issue.js:104]: You don't use labels here? or userName?, you only use title and url

File: components/core/marketing/Issue.js [hunk 9]
```diff
@@ -0,0 +1,119 @@
+import React, { Component } from "react";
```
Comment (jimmylee) [inline components/core/marketing/Issue.js:1]: Pretty much every file in the codebase does `import * as React from "react"`; so lets keep it all uniform.

File: components/core/marketing/IssuesList.js [hunk 10]
```diff
@@ -0,0 +1,34 @@
+import React, { Component } from "react";
+import * as Constants from "~/common/constants";
+
+import { css, keyframes } from "@emotion/react";
+
+import Issue from "~/components/core/marketing/Issue";
+
+const STYLES_SLATE_CARD_GROUP = css`
```
Comment (jimmylee) [inline components/core/marketing/IssuesList.js:8]: This style name isn't accurate.

File: pages/index.js [hunk 11]
```diff
@@ -582,7 +582,6 @@ export default class IndexPage extends React.Component {
     const response = await Actions.health();
     console.log("HEALTH_CHECK", response);
   }
```
Comment (jimmylee) [inline pages/index.js:584]: Why did you delete this line :P

**Review by jimmylee — CHANGES_REQUESTED** [review]
Pretty much every file in the codebase follows a certain formatting, so I would make those adjustments. The `STYLE_` names make no sense given the context you're using them, so I would update them (mentioned this in the PR).

Measurement nits are for you think about as you design/work with other pages. We are on a base 8 grid so stick to multiples of 4, 8, and 12 please.

File: components/core/marketing/Issue.js [hunk 12]
```diff
@@ -0,0 +1,119 @@
...
+  :hover,
+  :active {
+    color: ${Constants.system.pitchBlack};
+    background-color: transparent;
+  }
+  @media (max-width: ${Constants.sizes.mobile}px) {
+    font-size: 0.78rem;
+  }
+`;
+const STYLES_SLATE_CARD_CTA_PARAGRAPH = css`
```
Comment (jimmylee) [inline components/core/marketing/Issue.js:95]: You don't use this :)

File: components/core/marketing/Issue.js [hunk 13]
```diff
@@ -0,0 +1,119 @@
+import React, { Component } from "react";
+import * as Constants from "~/common/constants";
+
+import { css } from "@emotion/react";
+
+const STYLES_SLATE_CARD_GROUP = css`
```
Comment (jimmylee) [inline components/core/marketing/Issue.js:6]: You don't use this :)

File: components/core/marketing/Issue.js [hunk 14]
```diff
@@ -0,0 +1,119 @@
...
+  height: calc(100vh / 4);
+  margin: -1px 0 0 -1px;
+  transition: 200ms ease box-shadow;
+  border: 1px solid ${Constants.system.darkGray};
+  :hover {
+    transition: 200ms ease box-shadow;
+    box-shadow: 0px 10px 40px 20px rgba(0, 0, 0, 0.1);
+  }
+`;
+const STYLES_SLATE_CARD_CTA = css`
```
Comment (jimmylee) [inline components/core/marketing/Issue.js:23]: You don't use this :)

File: components/core/marketing/Issue.js [hunk 15]
```diff
@@ -0,0 +1,119 @@
...
+  padding: 12px;
+  font-size: ${Constants.typescale.lvl1};
+  text-align: left;
+  width: 100%;
+  @media (max-width: ${Constants.sizes.mobile}px) {
+    padding: 0px;
+    font-size: 1rem;
+  }
+`;
+const STYLES_SLATE_CARD_CTA_TITLE = css`
```
Comment (jimmylee) [inline components/core/marketing/Issue.js:55]: You don't use this :)

**Review by jimmylee — CHANGES_REQUESTED** [review]

File: components/core/marketing/CodeTerminal.js (hunk 3 above)
Comment (gndclouds) [reply components/core/marketing/CodeTerminal.js:8]: Can you clarify this particular formatting example, when Keyframes are used elsewhere it matches this format, for Example DataPicker.js so I think I'm missing something?

`const expand = keyframes`
  0% {
    max-height: 0px;
    overflow: hidden;
  }
  100% {
    max-height: 400px;
    overflow: hidden;
  }
`;`

Comment (jimmylee) [reply components/core/marketing/CodeTerminal.js:8]: Look at your spacing, look at the spacing in that example you copy and pasted

File: components/core/marketing/Issue.js (hunk 7 above)
Comment (gndclouds) [reply components/core/marketing/Issue.js:12]: Got it misunderstood the previous pr comment.

File: pages/index.js (hunk 11 above)
Comment (gndclouds) [reply pages/index.js:584]: Yep my bad I removed it from other pages and index by mistake

**Comment by gndclouds** [comment]
Ok, fixes addressed

File: components/core/ApplicationUserControls.js [hunk 16]
```diff
@@ -5,7 +5,7 @@ import * as SVG from "~/common/svg";
 import {
   TooltipWrapper,
   dispatchCustomEvent,
-  PopoverNavigation,
+  PopoverNavigation
```
Comment (jimmylee) [inline components/core/ApplicationUserControls.js:8]: We support trailing comma in prettier so maybe fix this one later

Comment (jimmylee) [reply components/core/ApplicationUserControls.js:8]: Also this file shouldn't be in your PR whats going on.

File: components/core/marketing/Issue.js [hunk 17]
```diff
@@ -0,0 +1,95 @@
+import * as React from "react";
+
+import * as Constants from "~/common/constants";
+import * as System from "~/components/system";
+
+import { css } from "@emotion/react";
+
+const STYLES_ISSUE_CARD_GROUP = css`
```
Comment (jimmylee) [inline components/core/marketing/Issue.js:8]: You don't use this :)

File: components/core/marketing/Issue.js [hunk 18]
```diff
@@ -0,0 +1,95 @@
+import * as React from "react";
+
```
Comment (jimmylee) [inline components/core/marketing/Issue.js:2]: If you're here, cut line 2

File: components/core/marketing/Issue.js [hunk 19]
```diff
@@ -0,0 +1,95 @@
...
+  }
+
+  @media (max-width: ${Constants.sizes.mobile}px) {
+    font-size: 0.78rem;
+  }
+`;
+
+export default class Issue extends React.Component {
+  render() {
+    const { title, id, url } = this.props;
```
Comment (jimmylee) [inline components/core/marketing/Issue.js:80]: you don't use id, and you can probably get away with `this.props.url` `this.props.title` and save yourself a line.

File: components/core/marketing/IssuesList.js [hunk 20]
```diff
@@ -0,0 +1,35 @@
+import * as React from "react";
+
```
Comment (jimmylee) [inline components/core/marketing/IssuesList.js:2]: You can cut line 2

File: pages/index.js [hunk 21]
```diff
@@ -581,7 +581,109 @@ export default class IndexPage extends React.Component {
   async componentDidMount() {
     const response = await Actions.health();
     console.log("HEALTH_CHECK", response);
-  }
+    this.addExampleSlates();
```
Comment (jimmylee) [inline pages/index.js:584]: Why did this come back?

File: pages/index.js [hunk 22]
```diff
@@ -581,7 +581,109 @@ export default class IndexPage extends React.Component {
...
+  }
+
+  coolSlates = [
```
Comment (jimmylee) [inline pages/index.js:587]: We had a zoom call about this, this shouldn't have come back

File: pages/index.js [hunk 23]
```diff
@@ -581,7 +581,109 @@ export default class IndexPage extends React.Component {
...
+                  <div css={STYLES_SLATE_CARD_PARAGRAPH}>{`@${c.slateUser}`}</div>
+                  <div css={STYLES_SLATE_CARD_PARAGRAPH}>-&gt;</div>
+                </div>
+              </div>
+            </a>
+          </div>
+        </div>
+      );
+    }
+    ReactDOM.render(allExampleSlates, document.getElementById("example-slates"));
```
Comment (jimmylee) [inline pages/index.js:684]: This whole function shouldn't have come back

**Review by jimmylee — CHANGES_REQUESTED** [review]
Almost :)

File: components/core/ApplicationUserControls.js (hunk 16 above)
Comment (gndclouds) [reply components/core/ApplicationUserControls.js:8]: Yea I was looking at it to make sure my code was structured correctly, and then couldnt remvoe it from the pr. (I think I clicked save and prettier made some change.)

File: components/core/marketing/Issue.js (hunk 19 above)
Comment (gndclouds) [reply components/core/marketing/Issue.js:80]: I was wondering about that, when would you, if ever, use `const { title, id, url } = this.props;` over just this.props.title for all the props?
